ansible/roles/sonicv2/tasks/sonic-cavm.yml

```diff
@@ -0,0 +1,58 @@
+# SONiC
+
+# Setup Platform
+#- include: platform-cavm.yml
+
+# Install Cavium Host Interface Driver
+- block:
+  - name: Rmmod for driver
+    modprobe: name=xp80-Pcie-Endpoint state=absent
+
+  - name: Install driver package
+    apt: pkg=xp80PcieEndpoint
+         state=present
+         force=yes
+
+  - name: Create a list of module dependencies
+    command: depmod -a
+
+  when: sonic_hwsku != "XP-SIM"
+  become: true
+  tags: drivermod
+
+# Setup port SKU
+- name: Setup port configuration file
+  become: yes
+  template: src=etc/ssw/{{ sonic_hwsku }}/profile.ini.j2
+            dest=/etc/ssw/{{ sonic_hwsku }}/profile.ini
+            owner=root
+            group=root
+            mode=0644
+  tags: syncd
+
+- name: Setup port configuration file
+  become: yes
+  shell: ln -s -f /etc/ssw/{{ sonic_hwsku }}/port_config_{{ sonic_portsku }}.ini /etc/ssw/{{ sonic_hwsku }}/port_config.ini
```

Comment:
Let me clarify my points. I guess your requirements are:
1. In the orchagent container, you will need a file "port_config.ini", and you will never read the files "port_config_*.ini" directly
2. The content of the file "port_config.ini" is determined during deployment, and never changed on the switch box.
Since the requirement is different from all other vendors. Suggest not to follow the "Copy SSW files".

To keep it very clean and consistent, suggest you not put source files in ansible/roles/sonicv2/files/ssw/. Instead, create a 'copy' task in sonic-cavm.yml, and the src parameter will be determined based on the port sku variable. So you don't need to copy many files to switch, don't need to create a symlink at all. Below is a sample (please test before using:))

    - name: Copy the port config file based on port sku
      copy:
        src: /etc/ssw_extra/{{ sonic_hwsku }}/port_config_{{ sonic_portsku }}.ini
        dest: /etc/ssw/{{ sonic_hwsku }}/port_config.ini



---
In reply to: [83538433](https://github.com/Azure/sonic-mgmt/pull/22#discussion_r83538433) [](ancestors = 83538433)